Cogmasters/concord
# Update dependencies, Bug fixes, OS detection

## What?
1. Replace `uthash.h` with `chash.h` and update `jsmn-find` version - See #62
2. Detect client OS - See 404d0a0258cf67ccc563192f211eaffce7c17d7d
3. **Bug Fixes**
3.1. Regression where `logconf.c` logging modules couldn't be disabled, therefore making `config.json` `disable_modules` field useless.
3.2. Uninitialized fields at `application_command.c` - See a2d89b2636e2e2984d94663e9272a6a85f338b64
3.3. Timer passing ID 0 - See #64

## Why?
1. See #62 for description
2. For Discord identification purposes, the previous behavior of defaulting to `"POSIX"` is a simplification that may end-up becoming a inaccuracy as we support more systems in the future.
3. **Bug Fixes**
3.1. Users reportedly were unable to use `discord_config_init()` on some builds (Android) due to a `logconf.c` regression where a `struct logconf` module and its sub-modules would each attempt to handle their `log_Logger *` uniquely, when in reality the same `log_Logger*` is shared amongst them all.
3.2. Would lead to a segfault at best, or access memory garbage at worst.
3.3. See #64 for description

## How?
1. See #62 for description
2. @HackerSmacker implementation's of `osname.h`, relying on the pre-processor to detect and assign the OS to a `OSNAME` macro.
3. **Bug Fixes**
3.1. Add a `is_disabled` boolean field to `struct logconf`, if enabled for a particular module then it shall be ignored for outputting.
3.2. Initialize `struct discord_request` necessary for handling the preliminary Client to Discord request context
3.3. See #64 for description

## Testing?
1. See #62 for description
2. Run under different OSes and check the output `http.log` for Discord Gateway's IDENTIFY payload.
3. **Bug Fixes**
3.1. Test `discord_config_init()` under valgrind for OOB detection.
3.2. Test under valgrind for OOB detection.
